models/shopify__customer_email_cohorts.sql

```diff
@@ -0,0 +1,69 @@
+with calendar as (
+
+    select *
+    from {{ ref('shopify__calendar') }}
+    where cast({{ dbt.date_trunc('month','date_day') }} as date) = date_day
+
+), customers as (
+
+    select *
+    from {{ ref('shopify__customer_emails') }}
+
+), orders as (
+
+    select *
+    from {{ ref('shopify__orders') }}
+
+), customer_calendar as (
+
+    select
+        calendar.date_day as date_month,
+        customers.email,
+        customers.first_order_timestamp,
+        customers.source_relation,
+        {{ dbt.date_trunc('month', 'first_order_timestamp') }} as cohort_month
+    from calendar
+    inner join customers
```

Comment:
Couldn't the `inner join` cause gaps in reporting if there are months with no customer email data? Unlikely but possible.

Reply:
yeah that'd be an issue if we were joining using a `=`, but since we are using `<=` this inner join should eliminate the appropriate chunks of data, i think

Reply:
I think if `customers` don't have any date data for a month (like say customers had data from October 2022 and December 2022 but not November 2022), then there would be no email cohort data from November because of the inner join rather than the left join since the inner join wouldn't recognize null dates.  

(Unless the `customers` table also has all the dates by default, in which case disregard). 

So in the cohort data you'd skip over a month because it'd be all null. When you'd probably just want zeroes to show up. Just want to make sure!

Reply:
ah so this join looks at the customers _first_ order timestamp. so if the customer made an order in october, but didn't make another order until december, november would still show up, as it is later than the customer's first order-timestamp

Reply:
Ahhhh nice! Yeah then I think the inner join should be good then.